src/codegen/sys/lib_.rs

```diff
@@ -376,6 +381,47 @@ fn generate_unions(w: &mut Write, env: &Env, items: &[&Union]) -> Result<()> {
     Ok(())
 }
 
+fn generate_debug_impl(w: &mut Write, name: &str, impl_content: &str) -> Result<()> {
+    writeln!(
+        w,
+        "impl ::std::fmt::Debug for {} {{\n\
+            \tfn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {{\n\
+                \t\t{}\n\
+            \t}}\n\
+        }}\n",
+        name,
+        impl_content)
+}
+
+fn generate_debug_with_fields(w: &mut Write, name: &str, lines: &[String]) -> Result<()> {
+    let fields =
+        lines.iter()
+             .filter_map(|field| {
+                 if field.contains("//") {
+                    None
+                 } else {
+                    let field_name = field.split(":").next().unwrap().trim().replace("pub ", "");
+                    Some(if field.replace(",", "").trim().ends_with("c_void") {
+                             format!("\t\t .field(\"{name}\", &\"c_void\")\n", name=field_name)
```

Comment:
Is really need write constant string in debug result?

Reply:
I meant: maybe better just skip this field?
Current results:
```rust
//gobject
impl ::std::fmt::Debug for GParamSpecString {
    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
        f.debug_struct(&format!("GParamSpecString @ {:?}", self as *const _))
         .field("parent_instance", &self.parent_instance)
         .field("default_value", &self.default_value)
         .field("cset_first", &self.cset_first)
         .field("cset_nth", &self.cset_nth)
         .field("substitutor", &self.substitutor)
         .field("_truncated_record_marker", &"c_void")
         .finish()
    }
}

//gtk
impl ::std::fmt::Debug for GtkRcStyle {
    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
        f.debug_struct(&format!("GtkRcStyle @ {:?}", self as *const _))
         .field("parent_instance", &self.parent_instance)
         .field("name", &self.name)
         .field("bg_pixmap_name", &self.bg_pixmap_name)
         .field("font_desc", &self.font_desc)
         .field("color_flags", &self.color_flags)
         .field("fg", &self.fg)
         .field("bg", &self.bg)
         .field("text", &self.text)
         .field("base", &self.base)
         .field("xthickness", &self.xthickness)
         .field("ythickness", &self.ythickness)
         .field("rc_properties", &self.rc_properties)
         .field("rc_style_lists", &self.rc_style_lists)
         .field("icon_factories", &self.icon_factories)
         .field("_truncated_record_marker", &"c_void")
         .finish()
    }
}

```

Reply:
I want all fields to be there. So better having a useless "c_void" string than nothing from my point of view.

Reply:
Ok